edk2toolext/invocables/edk2_ci_build.py

```diff
@@ -125,8 +134,41 @@ def GetLoggingFileName(self, loggerType: str) -> str:
         """Returns the filename (CI_BUILDLOG) of where the logs for the Edk2CiBuild invocable are stored in."""
         return "CI_BUILDLOG"
 
+    def ListPlugins(self) -> None:
+        """List all available plugins for the current configuration file and exit."""
+        pluginList = self.plugin_manager.GetPluginsOfClass(ICiBuildPlugin)
+
+        data = [
+            (entry.Module, str(pathlib.Path(entry.descriptor.get("descriptor_file", "unknown")).parent))
+            for entry in pluginList
+        ]
+
+        name_width = max(len("Plugin Name"), max(len(entry[0]) for entry in data))
+        dir_width = max(len("Plugin Directory (Code, Documentation)"), max(len(entry[1]) for entry in data))
+
+        header = f"{'Plugin Name'.ljust(name_width)} | {'Plugin Directory (Code, Documentation)'.ljust(dir_width)}"
+        separator = f"{'-' * name_width}-+-{'-' * dir_width}"
+
+        rows = []
+        for entry in data:
+            rows.append(f"{entry[0].ljust(name_width)} | {entry[1].ljust(dir_width)}")
+
+        table = "\n".join([header, separator] + rows)
+
+        print(table)
+
+        print("\nNOTE: You can disable any plugin with PluginName=skip.")
+        print(
+            "NOTE: You can disable all plugins with the --disable-all option, then enable specific plugins with "
+            "<PluginName>=run"
+        )
```

Comment:
You might also want to consider adding a note that says the plugins are not executed in the order shown.